packages/core/admin/admin/src/content-manager/components/DynamicZone/index.js

```diff
@@ -94,19 +94,11 @@ const DynamicZone = ({ name, labelAction, fieldSchema, metadatas }) => {
 
     const componentLayoutData = getComponentLayout(componentUid);
 
-    const allComponents = Object.values(dynamicComponentsByCategory).reduce((acc, components) => {
-      const componentObjects = components.reduce((acc, { componentUid, attributes }) => {
-        acc[componentUid] = {
-          attributes,
-          uid: componentUid,
-        };
-
-        return acc;
-      }, {});
-
-      return { ...acc, ...componentObjects };
-    }, {});
-
+    /**
+     * You have to pass _every component_ because the EditViewDataManager is not part of redux
+     * and you could have a dynamic component option that contains a component that is not part
+     * of the former list. Therefore it's schema is inaccessible leading to a crash.
+     */
```

Comment:
maybe it's implementation details, but what is a Category in the context of dynamicZones? I am trying to follow what logic was followed to generate dynamicComponentsByCategory before.

Reply:
A category is the category of the component, we display all dynamic zone component options grouped by their categories 👍🏼

Reply:
realistically when we move the content-manager back to redux this will be a lot cleaner because we can access the schemas in the reducer 🙃